src/psbt.c

```diff
@@ -4405,6 +4405,24 @@ static bool finalize_multisig(struct wally_psbt_input *input,
     return ret;
 }
 
+static bool finalize_p2tr(struct wally_psbt_input *input)
+{
+    struct wally_tx_witness_stack *final_witness;
+    const struct wally_map_item *sig;
+
+    sig = wally_map_get_integer(&input->psbt_fields, PSBT_IN_TAP_KEY_SIG);
+
+    /* TODO support tapleaf spends input->taproot_leaf_signatures */
+    if (!sig ||
+        wally_witness_keyspend_p2tr(sig->value, sig->value_len,
+                                    &final_witness) != WALLY_OK)
+        return false;
+
+    if (wally_psbt_input_set_final_witness(input, final_witness) != WALLY_OK)
+        return false;
+    return true;
```

Comment:
as above, if you generate directly you don't need to set this here.

Reply:
done